front/src/common/Map/Layers/MastLayers.tsx

```diff
@@ -0,0 +1,67 @@
+/**
+ * Shared layer showing the mast for icons on the map
+ */
+import { LayerProps, SymbolLayer } from 'react-map-gl/maplibre';
+import { isNil } from 'lodash';
+import React from 'react';
+import { OmitLayer, Theme } from 'types';
+import OrderedLayer from './OrderedLayer';
+
+export function getCommonSignsMastLayerProps(params: {
+  sourceTable?: string;
+  anglePropertyName?: string;
+  minzoom?: number;
+  maxzoom?: number;
+}): OmitLayer<SymbolLayer> {
+  const { anglePropertyName = 'angle', sourceTable } = params;
+  const res: OmitLayer<SymbolLayer> = {
+    type: 'symbol',
+    minzoom: 11,
+    maxzoom: 24,
+    paint: {},
+    layout: {
+      'text-field': '',
+      'text-font': ['Roboto Condensed'],
+      'text-size': 9,
+      'icon-image': [
+        'case',
+        ['==', ['get', 'side'], 'RIGHT'],
+        'MATD',
+        ['==', ['get', 'side'], 'LEFT'],
+        'MATG',
+        '',
+      ],
+      'icon-size': 0.7,
+      'icon-rotation-alignment': 'map',
+      'icon-pitch-alignment': 'map',
+      'icon-rotate': ['get', anglePropertyName],
+      'icon-allow-overlap': true,
+      'icon-ignore-placement': true,
+      'text-allow-overlap': true,
+      'text-ignore-placement': true,
+    },
+  };
+
+  if (!isNil(sourceTable)) {
+    res['source-layer'] = sourceTable;
+  }
+
+  return res;
+}
+
+interface MastLayerProps {
+  infraID: number | undefined;
+  colors: Theme;
+  layerOrder?: number;
+}
+
+export default function MastLayer(props: MastLayerProps) {
+  const { layerOrder } = props;
+
+  const mastsParams: LayerProps = getCommonSignsMastLayerProps({
+    sourceTable: 'neutral_signs',
```

Comment:
I this component is generic, `sourceTable` has to be a prop.

Reply:
made `sourceTable` a property